### PR TITLE
remove have_setuptools branching

### LINKED ISSUE #18113: BLD: since we already use setuptools, let's remove the optional logic in setup.py
no need to have the branching in setup.py we always have setuptools available.

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #18113
- [ ] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -363,8 +342,8 @@ def run(self):
         else:
             for pyxfile in self._pyxfiles:
                 cfile = pyxfile[:-3] + 'c'
-                msg = "C-source file '%s' not found." % (cfile) +\
-                    " Run 'setup.py cython' before sdist."
+                msg = ("C-source file '{cfile}' not found. Run "
+                       "'setup.py cython' before sdist.".format(cfile=cfile))
```
Comment (gfyoung) [inline setup.py]: :+1: - nice

**Comment by jbrockmendel** [comment]
I’m retroactively declaring this the first in a 4ish part sequence to transition to use cythonize.

File: setup.py [hunk 2]
```diff
@@ -38,46 +38,24 @@ def is_platform_mac():
...
 
-    setuptools_kwargs = {'zip_safe': False,
-                         'install_requires': ['python-dateutil >= 2',
-                                              'pytz >= 2011k',
-                                              'numpy >= %s' % min_numpy_ver],
-                         'setup_requires': ['numpy >= %s' % min_numpy_ver]}
-    if not _have_setuptools:
-        sys.exit("need setuptools/distribute for Py3k"
-                 "\n$ pip install distribute")
+setuptools_kwargs = {'test_suite': 'nose.collector',
```
Comment (jreback) [inline setup.py]: why did you add ``nose.collector``?

Comment (jreback) [reply setup.py]: I would completely remove this

**Review by jreback — CHANGES_REQUESTED** [review]

File: setup.py (hunk 2 above)
Comment (jbrockmendel) [reply setup.py]: > why did you add nose.collector?

It was already there, just lower in the file.

Comment (jreback) [reply setup.py]: I know, but I don't think its needed. remove and let's see.

Comment (jreback) [reply setup.py]: we *should* have a test_requires, but that's separate. can you open an issue for that.

**Comment by jbrockmendel** [comment]
Closing in favor of #18448.

**Comment by jreback** [comment]
thanks
